src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -67,6 +70,13 @@ export default class SourceFileLinter {
 				node as (ts.PropertyAccessExpression | ts.ElementAccessExpression)); // Check for deprecation
 		} else if (node.kind === ts.SyntaxKind.ImportDeclaration) {
 			this.analyzeImportDeclaration(node as ts.ImportDeclaration); // Check for deprecation
+		} else if (node.kind === ts.SyntaxKind.ExpressionWithTypeArguments) {
+			analyzeComponentJson(
```

Comment:
Just some idea on improving runtime performance:
Currently, this function is called for every class that extends from another class. Just to then go up the parent chain to find out that the file we're looking at is not a component. Wouldn't it be better to do the check whether we are looking at a Component at the time we are visiting the `SourceFile` node and then remember it for later? This would prevent the unnecessary overhead of always going up to the SourceFile node for every class.

Reply:
Done with https://github.com/SAP/ui5-linter/pull/73/commits/34f6698160a145eb4647905e4c489c1ac212a0c1